src/frontend/src/common/queries/groups.ts

```diff
@@ -250,3 +229,24 @@ export function useSendGroupInvitations({
     },
   });
 }
+
+/**
+ * expire all group-specific caches when group is changed in UI.
+ * this will not expire caches such as lists of locations
+ * or lineages because those won't change or vary depending on
+ * which group you are viewing
+ */
+export async function expireAllCaches(): Promise<void> {
+  const queriesToRefetch = [
+    USE_PHYLO_RUN_INFO,
+    USE_SAMPLE_INFO,
+    USE_GROUP_INFO,
+    USE_GROUP_INVITATION_INFO,
+    USE_GROUP_MEMBER_INFO,
+  ];
+
+  forEach(queriesToRefetch, async (q) => {
+    await queryClient.fetchQuery([q]);
+    await queryClient.invalidateQueries([q]);
```

Comment:
wouldn't you want to invalidate first before refetching?

Reply:
also won't it automatically refetch if you invalidate the queries?

Reply:
It does not always automatically refetch when you invalidate the queries. Invalidate refetches "smartly". React query has some internal logic to determine whether/when to refetch a query that has been invalidated. When you use refetch directly, the refetch is guaranteed. In this case, whatever smart thing react query was trying to do was preventing the refetch from happening, even though we wanted it to happen

Reply:
I do agree fetch should happen before invalidate. fixing...